src/CustomMarshalers/src/System/Runtime/InteropServices/CustomMarshalers/EnumeratorViewOfEnumVariant.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+
+namespace System.Runtime.InteropServices.CustomMarshalers
+{
+    [Serializable]
+    internal class EnumeratorViewOfEnumVariant : ICustomAdapter, IEnumerator
+    {
+        private readonly IEnumVARIANT m_enumVariantObject;
+        private bool m_fetchedLastObject;
+
+        public EnumeratorViewOfEnumVariant(IEnumVARIANT enumVariantObject)
+        {
+            m_enumVariantObject = enumVariantObject;
+            m_fetchedLastObject = false;
+            Current = null;
+
+        }
+
+        public object Current { get; private set; }
+
+        public unsafe bool MoveNext()
```

Comment:
Why is this `unsafe`?

Reply:
The last parameter of `System.Runtime.InteropServices.ComTypes.IEnumVARIANT.Next` is an `IntPtr` but really takes in a pointer to an int. So I use a local and take the address of the local to manually do the out-parameter. That requires unsafe.